Reorganised folders, Started cpu experiment metrics

Cerfoglg opened the pull request:
Organised folders for trials and experiments scripts. Added basic experiment metrics for cpu.

Works on #28, #22 
Fixes #25 
Fixes #21 
Fixes #19

Cerfoglg committed:
Reorganised folders, Started cpu experiment metrics

VincenzoFerme commented on analysers/experiment_analysers/cpu.py:
```diff
@@ -0,0 +1,75 @@
+
+
+# TODO: Fix this
+query = [{"experiment_id":experimentID, "cpu_mode":[modeMin, modeMax], "cpu_median":[medianMin, medianMax], \
```
I don't see an update to the Cassandra schema to store these data: https://github.com/benchflow/benchflow/pull/7/files

Cerfoglg committed:
Added integral trial metrics, dump size trial analyser

Cerfoglg committed:
Renamed folders for trials and experiments

VincenzoFerme commented:
@Cerfoglg please map in the description, the issues you worked on and the ones that are fixed by this pull request.

VincenzoFerme commented on analysers/experiments/cpu.py:
```diff
@@ -0,0 +1,87 @@
+cassandraHost = sys.argv[2]
+trialID = sys.argv[3]
+experimentID = trialID.split("_")[0]
+cassandraKeyspace = "benchflow"
```
`cassandraKeyspace`, `srcTable`, `destTable` should be defined as part of the configuration file and passed as parameters. The same applies also to other scripts.

Cerfoglg replied:
@VincenzoFerme This ties into the spark tasks sender issue of reorganising the way arguments are passed to the scripts

VincenzoFerme replied:
@Cerfoglg update that issue to point out you also need to update these scripts.

VincenzoFerme commented on analysers/experiments/cpu.py:
```diff
@@ -0,0 +1,87 @@
+        .first()
+    return v
+
+dataMin = sortAndGet("cpu_min", 1)
```
Why can't you just sort once and get all the data you need out of the sorted RDD?

VincenzoFerme commented on analysers/trials/cpu.py:
```diff
@@ -29,9 +32,26 @@
 
```
You should apply caching and configuration passed from the spark-tasks-sender also to the analysers script working on trials, and in general to all the scripts.

VincenzoFerme commented on analysers/trials/cpu.py:
```diff
@@ -29,9 +32,26 @@
-        return (r['cpu_percent_usage'], 0)
+        break
 
 data = sc.cassandraTable(cassandraKeyspace, srcTable) \
```
Why do you need to perform another query to Cassandra?

VincenzoFerme commented on analysers/trials/cpu.py:
```diff
@@ -77,12 +97,15 @@ def computeMedian(d):
 marginError = stdE * 2
 CILow = mean - marginError
 CIHigh = mean + marginError
+CI = [CILow, CIHigh]
```
Lets store them as two separated values

VincenzoFerme commented on analysers/trials/dumpSize.py:
```diff
@@ -0,0 +1,42 @@
+minioHost = sys.argv[4]
+filePath = sys.argv[5]
+experimentID = trialID.split("_")[0]
+cassandraKeyspace = "benchflow"
```
`cassandraKeyspace`, `table`, `minioPort` should be defined as part of the configuration file and passed as parameters. The same applies also to other scripts.

VincenzoFerme commented on analysers/trials/processDuration.py:
```diff
@@ -5,6 +5,9 @@
 import uuid
 import math
 
```
All the comments made for `analysers/cpu.py` apply

VincenzoFerme replied:
All the comments made for `analysers/cpu.py` apply

VincenzoFerme commented on analysers/trials/processDuration.py:
```diff
@@ -77,12 +97,15 @@ def computeMedian(d):
 CIHigh = mean + marginError
+CI = [CILow, CIHigh]
+
+dataIntegral = sum(integrate.cumtrapz(data))[0]
```
We don't need it

VincenzoFerme commented on analysers/trials/ram.py:
```diff
@@ -29,9 +29,26 @@
 
```
All the comments made for `analysers/cpu.py` apply

Cerfoglg committed:
Adjusted trial analysers

- Using numpy and scypy whenever possible
- Using correct format for trial tables to save to

Cerfoglg committed:
Adjusted format in cpu experiment analyser
